subsys/bootloader/include/fw_metadata.h

```diff
@@ -24,9 +24,17 @@
 #include <stddef.h>
 #include <toolchain.h>
 #include <assert.h>
+#include <string.h>
+#include <debug.h>
 
 #define MAGIC_LEN_WORDS (CONFIG_SB_MAGIC_LEN / sizeof(u32_t))
 
+#define TYPE_AND_DECL(retval, name, ...) \
```

Comment:
An explanation as to what this does an why it is needed would be great.

Reply:
Will add

Reply:
Changed the name to `EXTABI_FUNCTION` and added docs.